CMakeLists.txt

```diff
@@ -0,0 +1,599 @@
+check_symbol_exists(_SC_NPROCESSORS_CONF unistd.h TBLIS_HAVE__SC_NPROCESSORS_CONF)
+check_symbol_exists(_SC_NPROCESSORS_ONLN unistd.h TBLIS_HAVE__SC_NPROCESSORS_ONLN)
+
+# Find make
+find_program(MAKE_EXECUTABLE NAMES make gmake mingw32-make REQUIRED)
+
+set(ENABLE_SHARED_OLD ${ENABLE_SHARED})
+set(ENABLE_STATIC_OLD ${ENABLE_STATIC})
+set(ENABLE_SHARED OFF)
+set(ENABLE_STATIC ON)
+add_subdirectory(tblis/external/tci)
+add_subdirectory(tblis/external/marray)
+set(ENABLE_SHARED ${ENABLE_SHARED_OLD})
+set(ENABLE_STATIC ${ENABLE_STATIC_OLD})
+
+###############################################################################
+#
+# Find BLIS and set up plugin
+#
+###############################################################################
+
+file(STRINGS blis-git-tag BLIS_GIT_TAG)
+set(BLIS_INSTALL_DIR "${CMAKE_CURRENT_BINARY_DIR}/blis")
+
+FetchContent_Declare(blis
+    GIT_REPOSITORY https://github.com/flame/blis.git
+    GIT_TAG ${BLIS_GIT_TAG}
+    FIND_PACKAGE_ARGS 2.0...99.0
+)
+
+message(CHECK_START "Looking for BLIS")
+
+FetchContent_MakeAvailable(blis)
+FetchContent_GetProperties(
+  blis
+  SOURCE_DIR BLIS_SOURCE_DIR
+  BINARY_DIR BLIS_BINARY_DIR
+  POPULATED BLIS_POPULATED
+)
+
+set(BLIS_DEBUG_FLAG "")
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(BLIS_DEBUG_FLAG "--enable-debug=noopt")
+endif()
+
+if (BLIS_POPULATED)
+    message(CHECK_PASS "downloaded")
+
+    if (NOT EXISTS ${BLIS_BINARY_DIR}/config.mk)
+
+        # configure BLIS at CMake configure time
+        message(STATUS "Configuring BLIS")
+        execute_process(
+            WORKING_DIRECTORY ${BLIS_BINARY_DIR}
+            COMMAND ${BLIS_SOURCE_DIR}/configure
+                --prefix=${BLIS_INSTALL_DIR}
+                --disable-shared
+                --enable-static
+                ${BLIS_DEBUG_FLAG}
+                --disable-blas
+                --disable-cblas
+                "--enable-threading=pthread"
+                ${BLIS_CONFIG_FAMILY}
+        )
+        file(MAKE_DIRECTORY ${BLIS_INSTALL_DIR}/include/blis)
+    endif()
+
+    add_custom_command(
+        COMMAND ${MAKE_EXECUTABLE} install
+        OUTPUT
+            ${BLIS_INSTALL_DIR}/lib/libblis.a
+            ${BLIS_INSTALL_DIR}/include/blis/blis.h
+        COMMENT "Building BLIS"
+        WORKING_DIRECTORY ${BLIS_BINARY_DIR}
+    )
+    add_custom_target(blis-install
+        DEPENDS
+            ${BLIS_INSTALL_DIR}/lib/libblis.a
+            ${BLIS_INSTALL_DIR}/include/blis/blis.h
+    )
+else()
+    message(CHECK_PASS "found")
+    set(BLIS_BINARY_DIR ${BLIS_LIBDIR}/../share/blis)
+endif()
+
+# extract the blis configuration name (at CMake configure time) into BLIS_CONFIG_FAMILY
+execute_process(
+    INPUT_FILE ${BLIS_BINARY_DIR}/config.mk
+    COMMAND grep ^CONFIG_NAME
+    COMMAND awk "{ print \$3 }"
+    OUTPUT_VARIABLE BLIS_CONFIG_FAMILY
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+message(STATUS "BLIS configuration: ${BLIS_CONFIG_FAMILY}")
```

Comment:
```cmake
file(READ "${BLIS_BINARY_DIR}/config.mk" CONFIG_MK_CONTENTS)
string(REGEX MATCH "CONFIG_NAME[^\n]*" CONFIG_LINE "${CONFIG_MK_CONTENTS}")
string(REGEX MATCHALL "[^ \t]+" CONFIG_TOKENS "${CONFIG_LINE}")
list(GET CONFIG_TOKENS 2 BLIS_CONFIG_FAMILY)
message(STATUS "BLIS configuration: ${BLIS_CONFIG_FAMILY}")
```

Reply:
Thanks! It shouldn't be this hard to grep text...

Reply:
I agree...